spec.bs

```diff
@@ -1481,19 +1481,14 @@ null |winningComponentConfig|:
   1. Set |config| to |leadingBidInfo|'s [=leading bid info/auction config=].
   1. Let |sellerReportingScript| be the result of [=fetching script=] with |config|'s
     [=auction config/decision logic url=].
-  1. Let |reportResultOutputJS| be the result of [=evaluating a reporting script=] with
-     |sellerReportingScript|, "`reportResult`", and « |config|, |browserSignals|».
+  1. Let « |sellerSignals|, |reportUrl|, |reportingBeaconMap| » be the result of
+    [=evaluating a reporting script=] with |sellerReportingScript|, "`reportResult`", and
+    « |config|, |browserSignals|».
   1. [=map/Remove=] |browserSignals|["`dataVersion`"].
   1. [=map/Remove=] |browserSignals|["`modifiedBid`"].
   1. [=map/Remove=] |browserSignals|["`topLevelSellerSignals`"].
-  1. If |reportResultOutputJS| is an [=ECMAScript/abrupt completion=], return « null, |browserSignals| ».
-  1. Let |reportResultOutputIDL| be the result of [=converted to an IDL value|converting=]
-     |reportResultOutputJS| to a {{ReportResultOutput}}.
-  1. If an exception was [=exception/thrown=] in the previous step, return « null, |browserSignals| ».
-  1. TODO: Store |reportResultOutputIDL|["{{ReportResultOutput/reportUrl}}"] and
-     |reportResultOutputIDL|["{{ReportResultOutput/reportingBeaconMap}}"] in the
-     {{FencedFrameConfig}} as appropriate.
-  1. Return « |reportResultOutputIDL|["{{ReportResultOutput/signalsForWinner}}"], |browserSignals| ».
+  1. TODO: Store |reportUrl| and |reportingBeaconMap| in the {{FencedFrameConfig}} as appropriate.

```

Comment:
We're in parallel at this moment, right? Let's make this `[=fenced frame config=]` which is an infra struct, since I think we won't be constructing the platform object `{{FencedFrameConfig}}` _here_

Reply:
yes, we're in parallel, since its caller [=generate and score bids=] is in parallel. So we cannot construct IDL objects when it's in parallel? I wanted to define and construct a browserSignals IDL object (already did that for generateBid's, and scoreAd's is under review), if we cannot do that, we need to pass an ordered map to where we create a new |agent| and convert it to IDL object there? It adds more complexity.

Now I think I need to reconsider those in-parallels, for example, the [top level caller](https://github.com/WICG/turtledove/blob/main/spec.bs#L509). What I wanted to spec was that different component auctions can run in parallel without blocking each other, and script/signal fetches of different buyers can be run in parallel, etc., but it seems its scope is too wide.

Reply:
You can construct dictionaries in parallel. Not platform objects: https://webidl.spec.whatwg.org/#es-platform-objects